pkg/controller/blockdeviceclaim/blockdeviceclaim_controller_test.go

```diff
@@ -238,6 +254,121 @@ func (r *ReconcileBlockDeviceClaim) InvalidCapacityTest(t *testing.T,
 	r.CheckBlockDeviceClaimStatus(t, req, openebsv1alpha1.BlockDeviceClaimStatusInvalidCapacity)
 }
 
+func TestBlockDeviceClaimsLabelSelector(t *testing.T) {
+	// Set the logger to development mode for verbose logs.
+	logf.SetLogger(logf.ZapLogger(true))
+
+	tests := map[string]struct {
+		bdLabels           map[string]string
+		selector           *metav1.LabelSelector
+		expectedClaimPhase openebsv1alpha1.DeviceClaimPhase
+	}{
+		"only hostname label is present and no selector": {
+			bdLabels: map[string]string{
+				kubernetes.KubernetesHostNameLabel: fakeHostName,
+			},
+			selector:           nil,
+			expectedClaimPhase: openebsv1alpha1.BlockDeviceClaimStatusDone,
+		},
+		"no label is present and no selector": {
+			bdLabels:           map[string]string{},
+			selector:           nil,
+			expectedClaimPhase: openebsv1alpha1.BlockDeviceClaimStatusDone,
+		},
+		"custom label and hostname present on bd and selector": {
+			bdLabels: map[string]string{
+				ndm.KubernetesHostNameLabel: fakeHostName,
+				"ndm.io.test":               "1234",
```

Comment:
It will be good to add tests with values having alphabets & mix of alphabets, numbers & allowed special chars

Reply:
Is it okay if we take up in next PR, where we will refactor a bit of the list code and add more test cases. We have a client wrapper [here](https://github.com/openebs/node-disk-manager/blob/22ad376d653bf6a7831830c56c0554640fdca5ed/db/kubernetes/client.go#L130) and once a proper selector mechanism is added there, we will move all calls to that client.